packages/cli/src/tools/windows/androidWinHelpers.ts

```diff
@@ -0,0 +1,301 @@
+  /**
+   * The output of the following command is something like:
+   *
+   * ```
+   * accel:
+   * 0
+   * WHPX (10.0.19041) is installed and usable.
+   * accel
+   * ```
+   *
+   * If it fails it will still output to stdout with a similar format:
+   *
+   * ```
+   * accel:
+   * 1
+   * Android Emulator does not support nested virtualization.  Your VM host: 'Microsoft Hv' (Hyper-V)
+   * accel
+   * ```
+   *
+   */
+
+  try {
+    const {stdout} = await executeCommand(
+      `"${join(androidSDKRoot, 'emulator', 'emulator-check.exe')}" accel`,
+    );
+
+    return stdout;
+  } catch (e) {
+    const {stdout} = e;
+
+    return stdout;
+  }
+};
+
+/**
+ * Creates a new Android Virtual Device in the default folder with the
+ * name, device and system image passed by parameter.
+ */
+export const createAVD = async (
+  androidSDKRoot: string,
+  name: string,
+  device: string,
+  image: string,
+) => {
+  try {
+    const abi = image.includes('x86_64') ? 'x86_64' : 'x86';
+    const tag = image.includes('google_apis') ? 'google_apis' : 'generic';
+    const avdmanager = join(androidSDKRoot, 'tools', 'bin', 'avdmanager.bat');
+
+    const {stdout} = await executeCommand(
+      `${avdmanager} -s create avd --force --name "${name}" --device "${device}" --package "${image}" --tag "${tag}" --abi "${abi}"`,
+    );
+
+    // For some reason `image.sysdir.1` in `config.ini` points to the wrong location and needs to be updated
+    const configPath = join(
+      process.env.HOMEPATH || '',
+      '.android',
+      'avd',
+      `${name}.avd`,
+      'config.ini',
+    );
+
+    const content = await readFileAsync(configPath, 'utf-8');
+    const updatedContent = content.replace(
+      /Sdk\\system-images/g,
+      'system-images',
+    );
+
+    await writeFileAsync(configPath, updatedContent, 'utf-8');
+
+    return stdout;
+  } catch (e) {
+    const {stderr} = e;
+
+    return stderr;
+  }
+};
+
+/**
+ * Returns what hypervisor should be installed for the Android emulator
+ * using [Microsoft's official
+ * documentation](https://docs.microsoft.com/en-us/xamarin/android/get-started/installation/android-emulator/hardware-acceleration?pivots=windows)
+ * as a reference.
+ */
+export const getBestHypervisor = async (
+  androidSDKRoot: string,
+): Promise<HypervisorStatus> => {
+  // Should be in the path? might want to pass the root before
+  const customHypervisor = getProcessorType() === 'Intel' ? 'HAXM' : 'AMDH';
+
+  const stdout = await getEmulatorAccelOutputInformation(androidSDKRoot);
+
+  const lines = stdout.split('\n');
+
+  for (const line of lines) {
+    const hypervisor = parseHypervisor(line, customHypervisor);
+
+    if (hypervisor) {
+      return hypervisor;
+    }
+  }
+
+  // Couldn't identify the best one to run so not doing anything
+  return {
+    hypervisor: 'none',
+    installed: false,
+  };
+};
+
+export const enableWHPX = () => {
+  // Need to prompt for UAC
```

Comment:
Unsure what UAC is, but I assume it’s a comment that will be understood by people working in this area?